src/App.tsx

```diff
@@ -15,49 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+type Goods = string;
+type Reverse = boolean;
+
+enum Sortfield {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+type Props = {
+  goods: Goods[];
+};
+
+const GoodsList: React.FC<Props> = ({ goods }) => (
+  <ul>
+    {goods.map(good => (
+      <li data-cy="Good" key={good}>
+        {good}
+      </li>
+    ))}
+  </ul>
+);
+
+function getPrepareGoods(
+  goods: Goods[],
+  sortGood: Sortfield | undefined,
+  reverse: Reverse,
+): Goods[] {
+  const prepareGoods: Goods[] = [...goods];
+
+  if (prepareGoods) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortGood) {
+        case Sortfield.ALPHABET:
+          return good1.localeCompare(good2);
+
+        case Sortfield.LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortGood, setSortGood] = useState<Sortfield>(Sortfield.DEFAULT);
+  const [reverse, setReverse] = useState<Reverse>(false);
+  const visibleGoods = getPrepareGoods(goodsFromServer, sortGood, reverse);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortGood(Sortfield.ALPHABET)}
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-light': sortGood !== Sortfield.ALPHABET,
+          })}
         >
           Sort alphabetically
         </button>
-
         <button
+          onClick={() => setSortGood(Sortfield.LENGTH)}
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-light': sortGood !== Sortfield.LENGTH,
+          })}
         >
           Sort by length
         </button>
-
         <button
+          onClick={() => setReverse(!reverse)}
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', {
+            'is-light': reverse !== true,
+          })}
         >
           Reverse
         </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortGood || reverse) && (
+          <button
+            onClick={() => {
+              setSortGood(Sortfield.DEFAULT);
+              setReverse(false);
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
make handler in component and pass reference here